### PR TITLE
refactor: MdxInlineCode

### PR DESCRIPTION
**Pull request by zchsh**
## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

Moved `code` styles to an MDX `inlineCode` component, and cleans up related styles.

## 🤷 Why

- To avoid unintentional cascading of styles into custom components (such as `code` leaking into `@hashicorp/react-code-block`)
- To consolidate MDX styling in MDX components

## 🛠️ How

- removes `code` styles from `dev-dot-content`, moved to `MdxInlineCode` component
- removes `pre` style with border-radius
    - as far as I can tell, this style wasn't being applied - border radius in `@hashicorp/react-code-block` is applied on a different element, not `pre`
    - intent is to handle further refinements to `@hashicorp/react-code-block` in separate PRs
- removes `pre code` styles that reset aforementioned `code` styles within `@hashicorp/react-code-block`
    - use of `MdxInlineCode` prevents the unintentional cascade, so we no longer need to mitigate it

## 🧪 Testing

- Visit a docs MDX page, such as [/vault/docs/get-started/developer-qs][/vault/docs/get-started/developer-qs], and confirm code blocks and inline code looks as expected (vs [upstream][UPSTREAM/vault/docs/get-started/developer-qs])
- Visit a tutorial MDX page, such as [/waypoint/tutorials/get-started-docker/get-started-ui][/waypoint/tutorials/get-started-docker/get-started-ui], and confirm code blocks and inline code look as expected (vs [upstream][UPSTREAM/waypoint/tutorials/get-started-docker/get-started-ui])

## 💭 Anything else?

Not at the moment!

[task]: https://app.asana.com/0/0/1202072894754377/f
[preview]: https://dev-portal-git-zsconsolidate-mdx-code-css-hashicorp.vercel.app/
[/vault/docs/get-started/developer-qs]: https://dev-portal-git-zsconsolidate-mdx-code-css-hashicorp.vercel.app/vault/docs/get-started/developer-qs
[UPSTREAM/vault/docs/get-started/developer-qs]: https://developer.hashicorp.com/vault/docs/get-started/developer-qs
[/waypoint/tutorials/get-started-docker/get-started-ui]: https://dev-portal-git-zsconsolidate-mdx-code-css-hashicorp.vercel.app/waypoint/tutorials/get-started-docker/get-started-ui
[UPSTREAM/waypoint/tutorials/get-started-docker/get-started-ui]: https://developer.hashicorp.com/waypoint/tutorials/get-started-docker/get-started-ui

### CODE REVIEW

File: src/components/dev-dot-content/dev-dot-content.module.css [hunk 1]
```diff
@@ -57,29 +57,6 @@
...
-		border: 1px solid rgba(101, 106, 118, 0.2);
-		font-size: 0.8125rem;
-		font-weight: 400;
-		line-height: 16px;
-		padding: 2px 4px;
-	}
-
-	/* code blocks */
-	& pre {
-		border-radius: 5px;
```
Comment (zchsh) [inline src/components/dev-dot-content/dev-dot-content.module.css:73]: As far as I can tell, this style wasn't being applied - border radius in `@hashicorp/react-code-block` is applied on a different element, not `pre`. Broader intent here is to handle further refinements to `@hashicorp/react-code-block` in separate PRs.

Comment (ashleemboyer) [reply src/components/dev-dot-content/dev-dot-content.module.css:73]: This is probably from the beginning, before we were using CodeBlock 💭

File: src/components/dev-dot-content/dev-dot-content.module.css [hunk 2]
```diff
@@ -57,29 +57,6 @@
...
-
-		& code {
-			background-color: transparent;
-			border: none;
-			border-radius: 0;
-			padding: 0;
```
[the remark this answers is not in the document]
Comment (ashleemboyer) [reply src/components/dev-dot-content/dev-dot-content.module.css:79]: Same as above, I think this entire `& pre` block is from before we brought CodeBlock over. 💭

File: src/components/dev-dot-content/mdx-components/mdx-inline-code/mdx-inline-code.module.css [hunk 3]
```diff
@@ -0,0 +1,9 @@
+.inlineCode {
+	background-color: var(--token-color-palette-neutral-50);
+	border-radius: 5px;
+	border: 1px solid rgba(101, 106, 118, 0.2);
+	font-size: 0.8125rem;
+	font-weight: 400;
+	line-height: 16px;
+	padding: 2px 4px;
+}
```
Comment (ashleemboyer) [inline src/components/dev-dot-content/mdx-components/mdx-inline-code/mdx-inline-code.module.css:9]: Future 📌

We need to update some of these values to use HDS tokens, so I create a ticket to capture that work: [[MdxInlineCode] Update CSS to use hds tokens](https://app.asana.com/0/1202097197789424/1202578369060897/f) 🎟️

Comment (zchsh) [reply src/components/dev-dot-content/mdx-components/mdx-inline-code/mdx-inline-code.module.css:9]: 🙌 nice, ty!

**Pull request merged**